### DISCUSSION TITLE
Slim version of Debian images aren't updated

### DISCUSSION THREAD

**Discussion by mueller-ma** [renovatebot/renovate#24940] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

The codename detection of Debian version seems to fail on `*-slim` Docker images, e.g. `debian:bullseye-slim`.

Reproduced here: https://github.com/mueller-ma/renovate-debian-codename/pull/2

### Relevant debug logs

<details><summary>Logs</summary>

```
          {
            "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
            "currentValue": "bullseye-slim",
            "datasource": "docker",
            "depName": "debian",
            "depType": "final",
            "packageName": "debian",
            "replaceString": "debian:bullseye-slim",
            "skipReason": "invalid-value",
            "versioning": "docker",
            "warnings": [],
            "updates": []
          }
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
They are detected according to your logs. They're just not updated/update/able

**Reply by mueller-ma** [reply]
I updated the title of this discussion. The version should be updated as there's a `bookworm-slim` which is newer than `bullseye-slim`.

**Reply by HonkingGoose** [reply]
The [Renovate docs list a `debian` versioning](https://docs.renovatebot.com/modules/versioning/#debian-versioning). I suspect the `debian` versioning pulls its data from `https://debian.pages.debian.net/distro-info-data/debian.csv` because it's listed as a _reference_ in the docs.

Question for maintainers: would changing the versioning to `debian` for the `bookworm-slim` dependency help?

**Edit:** nevermind, the `debian` versioning docs says `-slim` is _not_ supported:

> Versions to which this scheme applies are e.g. `11` and `bullseye`. Container image tags such as `11-slim`, `11.4` or `11.4-slim`, `bullseye-20220822` or `bullseye-backports` are not compliant with the current implementation.

## Current `debian.csv`

version | codename | series | created | release | eol | eol-lts | eol-elts
-- | -- | -- | -- | -- | -- | -- | --
1.1 | Buzz | buzz | 1993-08-16 | 1996-06-17 | 1997-06-05 |   |  
1.2 | Rex | rex | 1996-06-17 | 1996-12-12 | 1998-06-05 |   |  
1.3 | Bo | bo | 1996-12-12 | 1997-06-05 | 1999-03-09 |   |  
2.0 | Hamm | hamm | 1997-06-05 | 1998-07-24 | 2000-03-09 |   |  
2.1 | Slink | slink | 1998-07-24 | 1999-03-09 | 2000-10-30 |   |  
2.2 | Potato | potato | 1999-03-09 | 2000-08-15 | 2003-07-30 |   |  
3.0 | Woody | woody | 2000-08-15 | 2002-07-19 | 2006-06-30 |   |  
3.1 | Sarge | sarge | 2002-07-19 | 2005-06-06 | 2008-03-30 |   |  
4.0 | Etch | etch | 2005-06-06 | 2007-04-08 | 2010-02-15 |   |  
5.0 | Lenny | lenny | 2007-04-08 | 2009-02-14 | 2012-02-06 |   |  
6.0 | Squeeze | squeeze | 2009-02-14 | 2011-02-06 | 2014-05-31 | 2016-02-29 |  
7 | Wheezy | wheezy | 2011-02-06 | 2013-05-04 | 2016-04-26 | 2018-05-31 | 2020-06-30
8 | Jessie | jessie | 2013-05-04 | 2015-04-26 | 2018-06-17 | 2020-06-30 | 2025-06-30
9 | Stretch | stretch | 2015-04-26 | 2017-06-17 | 2020-07-06 | 2022-06-30 | 2027-06-30
10 | Buster | buster | 2017-06-17 | 2019-07-06 | 2022-08-14 | 2024-06-30 | 2029-06-30
11 | Bullseye | bullseye | 2019-07-06 | 2021-08-14 | 2024-08-14 |   |  
12 | Bookworm | bookworm | 2021-08-14 | 2023-06-10 | 2026-06-10 |   |  
13 | Trixie | trixie | 2023-06-10 |   |   |   |  
14 | Forky | forky | 2025-08-01 |   |   |   |  
  | Sid | sid | 1993-08-16 |   |   |   |  
  | Experimental | experimental | 1993-08-16 |   |   |   |

**Reply by mueller-ma** [reply]
With the default settings `11-slim` *does* work, because `-slim` is recognized as platform/compatibility indicator: https://github.com/mueller-ma/renovate-debian-codename/pull/2/files

Maybe the implementation could ignore common suffixes like `-slim` and `-backports` for the Debian version before checking for updates.

**Reply by rarkins** [reply]
`-slim` (including `11-slim`) is not supported for _debian_ versioning, but it's supported for regular Docker vesioning. If you wish for Renovate to be debian-aware, then you need to use debian versioning. @HonkingGoose was correct

**Reply by HonkingGoose** [reply]
> With the default settings `11-slim` _does_ work, because `-slim` is recognized as platform/compatibility indicator: https://github.com/mueller-ma/renovate-debian-codename/pull/2/files

Good catch! The docs still says `11-slim` does _not_ work:

> Versions to which this scheme applies are e.g. `11` and `bullseye`. Container image tags such as `11-slim`, `11.4` or `11.4-slim`, `bullseye-20220822` or `bullseye-backports` are not compliant with the current implementation.

@rarkins can you suggest a different wording for this? I'll open a PR to update the text, if you want. 😉

**Reply by rarkins** [reply]
`11-slim` works with default docker versioning. It does not work with `debian` versioning, which is the page you copied from. The docs are correct

**Reply by HonkingGoose** [reply]
I'll keep the docs as they are now then. 😄

**Reply by rarkins** [reply, marked as answer]
In good news, this scenario will soon work thanks to #24717
